src/Shell/Task/ModelTask.php

```diff
@@ -344,25 +344,56 @@ public function findHasMany($model, array $associations)
                 }
             }
 
-            foreach ($otherSchema->columns() as $fieldName) {
-                $assoc = false;
-                if (!in_array($fieldName, $primaryKey) && $fieldName === $foreignKey) {
-                    $assoc = [
-                        'alias' => $otherModel->alias(),
-                        'foreignKey' => $fieldName
-                    ];
-                } elseif ($otherTableName === $tableName && $fieldName === 'parent_id') {
-                    $className = ($this->plugin) ? $this->plugin . '.' . $model->alias() : $model->alias();
-                    $assoc = [
-                        'alias' => 'Child' . $model->alias(),
-                        'className' => $className,
-                        'foreignKey' => $fieldName
-                    ];
+            $targetForeignKey = $primaryKey;
+            foreach ($targetForeignKey as $i => $fieldname) {
+                if ($fieldname === 'id') {
+                    $targetForeignKey[$i] = Inflector::singularize($model->table()) . '_id';
                 }
-                if ($assoc && $this->plugin && empty($assoc['className'])) {
-                    $assoc['className'] = $this->plugin . '.' . $assoc['alias'];
+            }
+
+            $possibleForeignKeys = $otherSchema->columns();
```

Comment:
Here are the Migrations (so you can easily pull the PR) and run
```
bin/cake bake all projects --force;
bin/cake bake all users --force;
bin/cake bake all groups --force;
bin/cake bake all tweets --force;
bin/cake bake all profiles --force;
bin/cake bake all comments --force;
bin/cake bake all dumb_comments --force;
```

# CreateUsers
```php
<?php
use Migrations\AbstractMigration;

class CreateUsers extends AbstractMigration
{
    /**
     * Change Method.
     *
     * More information on this method is available here:
     * http://docs.phinx.org/en/latest/migrations.html#the-change-method
     * @return void
     */
    public function change()
    {
        $table = $this->table('users', [
            'id' => false,
            'primary_key' => [
                'project_id',
                'id',
        ]]);
        $table->addColumn('project_id', 'integer', [
            'default' => null,
            'limit' => 11,
            'null' => false,
        ]);
        $table->addColumn('id', 'integer', [
            'autoIncrement' => true,
            'limit' => 11
        ]);
        $table->addColumn('name', 'string', [
            'default' => null,
            'limit' => 255,
            'null' => false,
        ]);
        $table->addForeignKey('project_id', 'projects', 'id', [
            'delete' => 'CASCADE',
            'update'=> 'NO_ACTION',
        ]);
        $table->create();
    }
}
```
# CreateGroups
```php
<?php
use Migrations\AbstractMigration;

class Groups extends AbstractMigration
{
    /**
     * Change Method.
     *
     * More information on this method is available here:
     * http://docs.phinx.org/en/latest/migrations.html#the-change-method
     * @return void
     */
    public function change()
    {
        $table = $this->table('groups', [
            'id' => false,
            'primary_key' => [
                'project_id',
                'id',
        ]]);
        $table->addColumn('id', 'integer', [
            'autoIncrement' => true,
            'limit' => 11
        ]);
        $table->addColumn('project_id', 'integer', [
            'default' => null,
            'limit' => 11,
            'null' => false,
        ]);
        $table->addColumn('name', 'string', [
            'default' => null,
            'limit' => 255,
            'null' => false,
        ]);
        $table->addForeignKey('project_id', 'projects', 'id', [
            'delete' => 'CASCADE',
            'update'=> 'NO_ACTION',
        ]);
        $table->create();
    }
}
```
# Create GroupsUsers
```php
<?php
use Migrations\AbstractMigration;

class CreateGroupsUsers extends AbstractMigration
{
    /**
     * Change Method.
     *
     * More information on this method is available here:
     * http://docs.phinx.org/en/latest/migrations.html#the-change-method
     * @return void
     */
    public function change()
    {
        $table = $this->table('groups_users', [
            'id' => false,
            'primary_key' => [
                'project_id',
                'group_id',
                'user_id',
        ]]);
        $table->addColumn('project_id', 'integer', [
            'default' => null,
            'limit' => 11,
            'null' => false,
        ]);
        $table->addColumn('group_id', 'integer', [
            'default' => null,
            'limit' => 11,
            'null' => false,
        ]);
        $table->addColumn('user_id', 'integer', [
            'default' => null,
            'limit' => 11,
            'null' => false,
        ]);
        $table->addForeignKey('project_id', 'projects', 'id', [
            'delete' => 'CASCADE',
            'update'=> 'NO_ACTION',
        ]);
        $table->addForeignKey(['project_id', 'group_id'], 'groups', ['project_id', 'id'], [
            'delete' => 'CASCADE',
            'update'=> 'NO_ACTION',
        ]);
        $table->addForeignKey(['project_id', 'user_id'], 'users', ['project_id', 'id'], [
            'delete' => 'CASCADE',
            'update'=> 'NO_ACTION',
        ]);
        $table->create();
    }
}
```
# CreateTweets
```php
<?php
use Migrations\AbstractMigration;

class CreateTweets extends AbstractMigration
{
    /**
     * Change Method.
     *
     * More information on this method is available here:
     * http://docs.phinx.org/en/latest/migrations.html#the-change-method
     * @return void
     */
    public function change()
    {
        $table = $this->table('tweets', [
            'id' => false,
            'primary_key' => [
                'project_id',
                'id',
        ]]);
        $table->addColumn('project_id', 'integer', [
            'default' => null,
            'limit' => 11,
            'null' => false,
        ]);
        $table->addColumn('id', 'integer', [
            'autoIncrement' => true,
            'limit' => 11
        ]);
        $table->addColumn('user_id', 'integer', [
            'default' => null,
            'limit' => 11,
            'null' => false,
        ]);
        $table->addColumn('body', 'text', [
            'default' => null,
            'null' => false,
        ]);
        $table->addForeignKey('project_id', 'projects', 'id', [
            'delete' => 'CASCADE',
            'update'=> 'NO_ACTION',
        ]);
        $table->addForeignKey(['project_id', 'user_id'], 'users', ['project_id', 'id'], [
            'delete' => 'CASCADE',
            'update'=> 'NO_ACTION',
        ]);
        $table->create();
    }
}
```
# CreateProfiles
```php
<?php
use Migrations\AbstractMigration;

class CreateProfiles extends AbstractMigration
{
    /**
     * Change Method.
     *
     * More information on this method is available here:
     * http://docs.phinx.org/en/latest/migrations.html#the-change-method
     * @return void
     */
    public function change()
    {
        $table = $this->table('profiles', [
            'id' => false,
            'primary_key' => [
                'project_id',
                'id',
        ]]);
        $table->addColumn('project_id', 'integer', [
            'default' => null,
            'limit' => 11,
            'null' => false,
        ]);
        $table->addColumn('id', 'integer', [
            'autoIncrement' => true,
            'limit' => 11
        ]);
        $table->addColumn('user_id', 'integer', [
            'default' => null,
            'limit' => 11,
            'null' => false,
        ]);
        $table->addColumn('nickname', 'string', [
            'default' => null,
            'null' => false,
        ]);
        $table->addForeignKey('project_id', 'projects', 'id', [
            'delete' => 'CASCADE',
            'update'=> 'NO_ACTION',
        ]);
        $table->addForeignKey(['project_id', 'user_id'], 'users', ['project_id', 'id'], [
            'delete' => 'CASCADE',
            'update'=> 'NO_ACTION',
        ]);
        $table->addIndex(['project_id', 'user_id'], ['unique' => true]);
        $table->create();
    }
}
```
# CreateComments
```php
<?php
use Migrations\AbstractMigration;

class CreateComments extends AbstractMigration
{
    /**
     * Change Method.
     *
     * More information on this method is available here:
     * http://docs.phinx.org/en/latest/migrations.html#the-change-method
     * @return void
     */
    public function change()
    {
        $table = $this->table('comments', [
            'id' => false,
            'primary_key' => [
                'project_id',
                'id',
        ]]);
        $table->addColumn('project_id', 'integer', [
            'default' => null,
            'limit' => 11,
            'null' => false,
        ]);
        $table->addColumn('id', 'integer', [
            'autoIncrement' => true,
            'limit' => 11
        ]);
        $table->addColumn('user_id', 'integer', [
            'default' => null,
            'limit' => 11,
            'null' => false,
        ]);
        $table->addColumn('tweet_id', 'integer', [
            'default' => null,
            'limit' => 11,
            'null' => false,
        ]);
        $table->addColumn('body', 'text', [
            'default' => null,
            'null' => false,
        ]);
        $table->addForeignKey('project_id', 'projects', 'id', [
            'delete' => 'CASCADE',
            'update'=> 'NO_ACTION',
        ]);
        $table->addForeignKey(['project_id', 'user_id'], 'users', ['project_id', 'id'], [
            'delete' => 'CASCADE',
            'update'=> 'NO_ACTION',
        ]);
        $table->addForeignKey(['project_id', 'tweet_id'], 'tweets', ['project_id', 'id'], [
            'delete' => 'CASCADE',
            'update'=> 'NO_ACTION',
        ]);
        $table->create();
    }
}
```
# Create DumbComments
```php
<?php
use Migrations\AbstractMigration;

class CreateDumbComments extends AbstractMigration
{
    /**
     * Change Method.
     *
     * More information on this method is available here:
     * http://docs.phinx.org/en/latest/migrations.html#the-change-method
     * @return void
     */
    public function change()
    {
        $table = $this->table('dumb_comments', [
            'id' => false,
            'primary_key' => [
                'project_id',
                'id',
        ]]);
        $table->addColumn('project_id', 'integer', [
            'default' => null,
            'limit' => 11,
            'null' => false,
        ]);
        $table->addColumn('id', 'integer', [
            'autoIncrement' => true,
            'limit' => 11
        ]);
        $table->addColumn('user_id', 'integer', [
            'default' => null,
            'limit' => 11,
            'null' => false,
        ]);
        $table->addColumn('tweet_id', 'integer', [
            'default' => null,
            'limit' => 11,
            'null' => false,
        ]);
        $table->addColumn('body', 'text', [
            'default' => null,
            'null' => false,
        ]);
        $table->create();
    }
}
```